source/kernel/src/services/emb_display.rs

```diff
@@ -137,81 +103,244 @@ impl EmbDisplayClient {
+/// color and transparency depths.
+///
+/// Users may use any kind of buffer they'd like, and displays are expected to
+/// convert to a format that they can use, for example down or upsampling color
+/// depth, converting color to grayscale, etc.
+///
+/// This [FrameChunk] is passed to [EmbDisplayClient::draw()] to be rendered to
+/// the display
+///
+/// ## Subsizing
+///
+/// This frame chunk is expected to be equal or smaller than the total display
+/// itself, and each of the kinds of framechunk will have [metadata] that contains
+/// both the position and the size of the framechunk. Framechunks can be moved
+/// but not resized (if resizing is necessary, the current chunk should be dropped
+/// and a new one should be allocated).
+///
+/// [metadata]: [FrameBufMeta]
+///
+/// ## Transparency
+///
+/// FrameChunks also have a transparency component, which allows them to be used for
+/// [blitting] or [compositing] onto the final display image.
+///
+/// [blitting]: https://en.wikipedia.org/wiki/Bit_blit
+/// [compositing]: https://en.wikipedia.org/wiki/Compositing
+///
+/// For example, a circle with radius of 80 pixels could be drawn in a 100x100
+/// FrameChunk, with the area outside of the circle marked as transparent. This
+/// allows the 100x100 square to be "blitted" at the target location without
+/// overwriting the existing content or background outside the circle.
+///
+/// This could also be used to keep persistently drawn [sprites] in memory,
+/// layering them onto the target frame.
+///
+/// All FrameChunk kinds have some kind of transparency, though this may range from
+/// a single bit transparency (transparent or not), to a more complex [alpha channel]
+///
+/// [alpha channel]: https://en.wikipedia.org/wiki/Alpha_compositing
+/// [sprites]: https://en.wikipedia.org/wiki/Sprite_(computer_graphics)
+#[non_exhaustive]
+pub enum FrameChunk {
+    Mono(MonoChunk),
+}
+
+impl From<MonoChunk> for FrameChunk {
+    fn from(value: MonoChunk) -> Self {
+        FrameChunk::Mono(value)
+    }
+}
+
+// TODO: both the data and the mask could be stored 1bpp, however because
+// that math was beyond me at the time, I am storing them 8bpp, which is
+// very wasteful in terms of memory, but means that we don't need to do
+// tricky bit operations.
+//
+// On the other hand, it is likely a bit less computationally intense to
+// stick with byte addressing, as we don't need to do shifting and such
+// for individual pixel operations, but there might be nice ways to accelerate
+// that, though then you need to worry about "alignment" of data, e.g. if the
+// start_x is not a multiple of 8.
+//
+// It may just be worth adding a "MonoChunk1bpp" variant in the future to allow
+// users to make the size/perf tradeoff, particularly if we want to support
+// targets with very small memory. For example, a 400x240 monochrome display would
+// be 93.75KiB at 8bpp, but only 11.72KiB at 1bpp.
```

Comment:
taking a `bitvec` dep probably lets us do 1bpp without having to think too hard about How To Do Tricky Bit Operations, and @myrrlyn is cool...

Reply:
Yeah, I think I mentioned the same in chat, and meant to put it in the todo here, but I also think it's a very good idea. We mostly just need to make `mnemos_alloc::containers` versions of `BitVec` (or likely rather `BitArr`), and I think the bitvec crate has traits for that.